cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
@sumwale Regarding the later question,  multiple concurrent streaming jobs will create multiple SnappyStreaming Context objects just like multiple normal jobs will create multiple SnappySession objects. Each job ( hence each object)  will add their respective dependent URLs in their objects.